tools/pipelines/templates/include-telemetry-setup.yml

```diff
@@ -1,146 +1,91 @@
-    - task: Npm@1
-      displayName: 'npm install aria logger'
-      inputs:
-        command: 'custom'
-        workingDir: ${{ parameters.testWorkspace }}
-        customCommand: 'install ${{ parameters.loggerPackage }}'
-        customRegistry: 'useNpmrc'
-
-    - task: Bash@3
-      displayName: 'Prepare telemetry-generator'
-      inputs:
-        targetType: 'inline'
-        workingDirectory: $(toolAbsolutePath)
-        script: |
-          cp ${{ parameters.testWorkspace }}/.npmrc . ;
-          npm install @ff-internal/aria-logger;
-          npm i;
-          npm run build:compile;
+# URL for the "Office feed" in our ADO project. Packages in the @microsoft scope will be resolved from here when
+# installing the dependencies for telemetry generator and aria-logger.
+- name: officeFeedUrl
+  type: string
 
-    - ${{ parameters.customSteps }}
+steps:
+# Need to checkout the repo in order to run @fluid-tools/telemetry-generator which we don't publish right now.
+- checkout: self
+  clean: true
+
+- template: include-use-node-version.yml
+
+- task: Bash@3
+  displayName: Print parameter/variable values for template
+  inputs:
+    targetType: 'inline'
+    script: |
+      echo "
+      Parameters:
+        pathToTelemetryGenerator=${{ parameters.pathToTelemetryGenerator }}
+        devFeedUrl={{ parameters.devFeedUrl }}
+        officeFeedUrl={{ parameters.officeFeedUrl }}
+      "
+
+- task: Bash@3
+  displayName: Initialize npmrc
+  inputs:
+    targetType: 'inline'
+    workingDirectory: ${{ parameters.pathToTelemetryGenerator }}
+    # Note: $(ado-feeds-build) and $(ado-feeds-office) come from the ado-feeds variable group
+    script: |
+      echo Initialize package
+      npm init --yes
+
+      echo Generating .npmrc
+      echo "registry=https://registry.npmjs.org" >> ./.npmrc
+      echo "always-auth=false" >> ./.npmrc
+
+      echo "@fluidframework:registry=${{ parameters.devFeedUrl }}" >> ./.npmrc
+      echo "@fluid-experimental:registry=${{ parameters.devFeedUrl }}" >> ./.npmrc
+      echo "@fluid-tools:registry=${{ parameters.devFeedUrl }}" >> ./.npmrc
+      echo "@fluid-internal:registry=${{ parameters.devFeedUrl }}" >> ./.npmrc
+      echo "@fluid-private:registry=${{ parameters.devFeedUrl }}" >> ./.npmrc
+      echo "@ff-internal:registry=${{ parameters.devFeedUrl }}" >> ./.npmrc
+      echo "@microsoft:registry=${{ parameters.officeFeedUrl }}" >> ./.npmrc
+      echo "always-auth=true" >> ./.npmrc
+      cat .npmrc
+
+# Auth to internal ADO feeds
+- task: npmAuthenticate@0
+  displayName: 'npm authenticate (internal ADO feeds)'
+  inputs:
+    workingFile: ${{ parameters.pathToTelemetryGenerator }}/.npmrc
+
+# Install logger package
+- task: Npm@1
+  displayName: 'npm install aria logger'
+  inputs:
+    workingDir: ${{ parameters.pathToTelemetryGenerator }}
+    command: 'custom'
+    customCommand: 'install @ff-internal/aria-logger'
+    customRegistry: 'useNpmrc'
+
+- task: Bash@3
```

Comment:
Nit: would this be better off as 2 separate `Npm@1` tasks?

Reply:
Yeah, I can see that. I'll create a task to address that later as low-pri; it takes a while to get all pipelines to complete for validating any given commit, and the runs for the current HEAD of this branch are close to completing 😄 .